packages/babel-parser/src/plugins/flow.js

```diff
@@ -2721,17 +2723,29 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     }
 
     skipFlowComment(): number | boolean {
-      const ch2 = this.input.charCodeAt(this.state.pos + 2);
-      const ch3 = this.input.charCodeAt(this.state.pos + 3);
+      let firstNonWhiteSpace = this.state.pos + 2;
+      while (
+        [charCodes.space, charCodeTab].includes(
+          this.input.charCodeAt(firstNonWhiteSpace),
+        )
+      ) {
+        firstNonWhiteSpace++;
+      }
+
+      const ch2 = this.input.charCodeAt(firstNonWhiteSpace);
+      const ch3 = this.input.charCodeAt(firstNonWhiteSpace + 1);
 
       if (ch2 === charCodes.colon && ch3 === charCodes.colon) {
-        return 4; // check for /*::
+        return firstNonWhiteSpace + 2; // check for /*::
```

Comment:
I think this was what the big issue was. `this.state.pos` was already added, so it was returning the index to shift to instead of the amount to shift by.